### PR TITLE
Release from CI

### PR DESCRIPTION
**Pull request by alexarchambault**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,5 +1,9 @@
 sudo: false
 language: scala
+stages:
+  - name: test
+  - name: release
+    if: (branch = master AND type = push) OR (tag IS present)
 matrix:
```
Comment (olafurpg) [inline .travis.yml]: I think this needs to be `stages` to use stages

File: .travis.yml [hunk 2]
```diff
@@ -26,6 +30,12 @@ matrix:
   - env: CMD="mill demo.fullOpt && sbt readme/run"
     jdk: oraclejdk8
 
-script:
+  - stage: release
+    script:
+      - echo "$PGP_SECRET" | base64 --decode | gpg --import
+      - mill __.publish --sonatypeCreds "$SONATYPE_USERNAME:$SONATYPE_PASSWORD" --release true --gpgPassphrase "$PGP_PASSPHRASE"
```
Comment (olafurpg) [inline .travis.yml]: Will this run on every merge into master or only for tag pushes?

Comment (alexarchambault) [reply .travis.yml]: Every merge. But things are automatically pushed to the right repo (releases or snapshots).

File: .travis.yml (hunk 1 above)
Comment (alexarchambault) [reply .travis.yml]: You mean `jobs`? I was under the impression that `jobs` and `matrix` were equivalent (and [this](https://travis-ci.org/lihaoyi/fastparse/builds/524877277) goes in that sense).

Comment (alexarchambault) [reply .travis.yml]: Just renamed it to `jobs` just in case.

File: .travis.yml [hunk 3]
```diff
@@ -1,6 +1,10 @@
 sudo: false
 language: scala
-matrix:
+stages:
+  - name: test
+  - name: release
+    if: (branch = master AND type = push) OR (tag IS present)
+jobs:
```
Comment (olafurpg) [inline .travis.yml]: :+1:

File: .travis.yml (hunk 1 above)
Comment (olafurpg) [reply .travis.yml]: I meant `jobs` yes, but maybe it's equivalent to `matrix`, I don't know 🤔

**Comment by olafurpg** [comment]
Let's see if the snapshot release works 🍿 https://travis-ci.org/lihaoyi/fastparse/jobs/524888327

**Comment by olafurpg** [comment]
Looks like a success 🎉 

```
coursier fetch com.lihaoyi:fastparse_2.13.0-RC1:2.1.0-8+aae1dca-SNAPSHOT -r sonatype:snapshots
```

**Comment by alexarchambault** [comment]
✨ on first try ✨

**Comment by olafurpg** [comment]
The release for 2.1.1 timed out on travis 😭  https://travis-ci.org/lihaoyi/fastparse/jobs/524928811

Should we try to split the release stage into parallel jobs for 2.11,2.12,2.13?

**Pull request merged**